api/envoy/api/v2/lds.proto

```diff
@@ -57,10 +58,21 @@ message Listener {
   // :ref:`FilterChainMatch <envoy_api_msg_listener.FilterChainMatch>` criteria is used on a
   // connection.
   //
+  // Precisely one of filter_chains and fdcs_config must be set.
+  //
   // Example using SNI for filter chain selection can be found in the
   // :ref:`FAQ entry <faq_how_to_setup_sni>`.
-  repeated listener.FilterChain filter_chains = 3
-      [(validate.rules).repeated .min_items = 1, (gogoproto.nullable) = false];
+  repeated listener.FilterChain filter_chains = 3 [(gogoproto.nullable) = false];
+
+  message FcdsConfig {
+    core.ConfigSource config = 1 [(validate.rules).message.required = true];
+
+    // Optional alternative to the listener name to present to FCDS.
+    string filter_chain_name = 2;
```

Comment:
I was trying to copy the semantics of EDS for consistency.

Are you suggesting that the filter chains should be entirely independent of the listeners? That would allow multiple listeners to share a set of filter-chains, but we'd have to think carefully about the lifetime issues.

Reply:
I agree with @htuch . FilterChains should be the resource that is being requested (mandatory and never optional). We can combine it with the listener name for precision. I am not sure sharing filter chains makes sense in the common case

Reply:
If we did that, it would no longer solve the original problem: we want to be able to add/remove filter chains without modifying the listener (which would result in listener draining).

The answer may be to support both modes: either request "all filter chains for this listener", or request "filter chain X for this listener".

Won't most/all of the xDS APIs need to be augmented in a similar way to support incremental?

Reply:
As discussed offline, it seems reasonable to make listener changes non-draining for the filter chain list.